File: 02._Data_Files/01a_File_formats_bonanza.js
const fs = require('fs');
const yaml = require('js-yaml');
//const xml2js = require('xml2js');
const xmlParse = require('fast-xml-parser');
const csv = require('csv-parser');

//Text file read
const textContent = fs.readFileSync('./me.text', 'utf-8');
const itemsFromText = textContent.split('\n').map(item => item.trim());
console.log('Text File Content:', itemsFromText);

//json file read
const jsonContent = fs.readFileSync('./me.json', 'utf-8');
const itemsFromJSON = JSON.parse(jsonContent);
console.log('JSON File Content:', itemsFromJSON);

//yaml file read
const yamlContent = fs.readFileSync('./me.yaml', 'utf-8');
const itemsFromYAML = yaml.load(yamlContent);
console.log('YAML File Content:', itemsFromYAML);

//xml file read xml2js issues - xml-fast-parser used
const xmlContent = fs.readFileSync('./me.xml', 'utf-8');
const parser = new xmlParse.XMLParser();
const itemsFromXML = parser.parse(xmlContent);
console.log('XML File Content:', itemsFromXML);

//csv file read
const itemsFromCSV = [];
fs.createReadStream('./me.csv')
  .pipe(csv())
  .on('data', (row) => {
    // Assuming 'name' is the column containing the items
    itemsFromCSV.push(row);
  })
  .on('end', () => {
    console.log('CSV File Content:', itemsFromCSV);
  });

File: Assignments/01._Data_Files/01a_File_formats_bonanza.py
import xml.etree.ElementTree as ET
import yaml
import json
import csv

#Python read file
#Text file
with open('./me.text', 'r') as file:
    items_from_text = [line.strip() for line in file.readlines()]

print('Text File Content:', items_from_text)


#xml file read
tree = ET.parse('./me.xml')
root = tree.getroot()
data = {}
for child in root:
    if len(child) > 0:
        data[child.tag] = [grandchild.text for grandchild in child]
    else:
        data[child.tag] = child.text

print('XML File Content: ', data)

#yaml file read
with open('./me.yaml', 'r') as file:
    items_from_yaml = yaml.safe_load(file)

print('YAML File Content:', items_from_yaml)


#JSON file read
with open('./me.json', 'r') as file:
    items_from_json = json.load(file)

print('JSON File Content:', items_from_json)


#CSV file read
items_from_csv = []
with open('./me.csv', 'r') as file:
    csv_reader = csv.reader(file)
    for row in csv_reader:
        items_from_csv.append(row)

print('CSV File Content:', items_from_csv)

